abcclassroom/config.py

```diff
@@ -0,0 +1,96 @@
+"""
+abc-classroom.config
+====================
+
+"""
+
+import os
+import sys
+
+import os.path as op
+
+from ruamel.yaml import YAML
+
+
+def get_github_auth():
+    """
+    Check to see if there is an existing github authentication
+    and load the authentication.
+
+    Returns
+    -------
+    ruamel.yaml.comments.CommentedMap
+        Yaml object that contains the token and id for a github session.
+        If yaml doesn't exists, return an empty dictionary.
+    """
+    yaml = YAML()
+    try:
+        with open(op.expanduser("~/.abc-classroom.tokens.yml")) as f:
+            config = yaml.load(f)
+        return config["github"]
+
+    except FileNotFoundError:
+        return {}
+
+
+def set_github_auth(auth_info):
+    """
+    Set the github authentication information. Put the token and id authentication
+    information into a yaml file if it doesn't already exist.
+
+    Parameters
+    ----------
+    auth_info : dictionary
+        The token and id authentication information from github stored in a
+        dictionary object.
+    """
+    yaml = YAML()
+    config = {}
+    if get_github_auth():
+        with open(op.expanduser("~/.abc-classroom.tokens.yml")) as f:
+            config = yaml.load(f)
+
+    config["github"] = auth_info
+
+    with open(op.expanduser("~/.abc-classroom.tokens.yml"), "w") as f:
+        yaml.dump(config, f)
+
+
+def get_config():
+    yaml = YAML()
+    try:
+        with open("config.yml") as f:
+            config = yaml.load(f)
+        return config
+    except FileNotFoundError as err:
+        print(
+            "Error: config file (config.yml) not found in current directory\n"
```

Comment:
i ran into this error because i had sample_config.yml from nathan's quickstart. 
I wonder if this can be a friendlier message?

```
Oops! I need a config.yml file in order to setup your github classroom environment. Are you in your class directory? If you don't have a class directory yet, create one using  abc-quickstart.
```

Reply:
ok so after i fixed the config file name i got this
```
(abc-dev) CIRES-EL-LM-020:my-course leahwasser$ abc-assignment-template sdfs
Loading configuration from config.yml
Traceback (most recent call last):
  File "/Users/leahwasser/miniconda3/envs/abc-dev/bin/abc-assignment-template", line 11, in <module>
    load_entry_point('abc-classroom', 'console_scripts', 'abc-assignment-template')()
  File "/Users/leahwasser/Documents/github/0-python/abc-classroom/abcclassroom/__main__.py", line 433, in assignment_template
    config = cf.get_config()
  File "/Users/leahwasser/Documents/github/0-python/abc-classroom/abcclassroom/config.py", line 63, in get_config
    config = yaml.load(f)
  File "/Users/leahwasser/miniconda3/envs/abc-dev/lib/python3.7/site-packages/ruamel/yaml/main.py", line 341, in load
    return constructor.get_single_data()
  File "/Users/leahwasser/miniconda3/envs/abc-dev/lib/python3.7/site-packages/ruamel/yaml/constructor.py", line 111, in get_single_data
    node = self.composer.get_single_node()
  File "/Users/leahwasser/miniconda3/envs/abc-dev/lib/python3.7/site-packages/ruamel/yaml/composer.py", line 78, in get_single_node
    document = self.compose_document()
  File "/Users/leahwasser/miniconda3/envs/abc-dev/lib/python3.7/site-packages/ruamel/yaml/composer.py", line 101, in compose_document
    node = self.compose_node(None, None)
  File "/Users/leahwasser/miniconda3/envs/abc-dev/lib/python3.7/site-packages/ruamel/yaml/composer.py", line 138, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/Users/leahwasser/miniconda3/envs/abc-dev/lib/python3.7/site-packages/ruamel/yaml/composer.py", line 218, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/Users/leahwasser/miniconda3/envs/abc-dev/lib/python3.7/site-packages/ruamel/yaml/composer.py", line 136, in compose_node
    node = self.compose_sequence_node(anchor)
  File "/Users/leahwasser/miniconda3/envs/abc-dev/lib/python3.7/site-packages/ruamel/yaml/composer.py", line 180, in compose_sequence_node
    node.value.append(self.compose_node(node, index))
  File "/Users/leahwasser/miniconda3/envs/abc-dev/lib/python3.7/site-packages/ruamel/yaml/composer.py", line 116, in compose_node
    None, None, 'found undefined alias %r' % utf8(alias), event.start_mark
ruamel.yaml.composer.ComposerError: found undefined alias '.ipynb'
  in "config.yml", line 40, column 3
```

something with my config is off... will post a copy of it in a minute.

Reply:
```

# The GitHub organization to host your GitHub classroom
# Create a new organization on GitHub and change this to the name you gave it
organization: earth-analytics-edu

# (Optional) Short name for organization. If set, this gets used to prepend
# the repositories instead of the full organization name, just to make
# names shorter and less unwieldy
organization-shortname: ea

# Name of your course. Will be used to prefix repositories inside the
# organization. The repositories will be named:
# https://github.com/<organization>/<courseName>-template
# https://github.com/<organization>/<courseName>-<studentName>
# or, if organization-shortname is set:
# https://github.com/<organization-shortname>/<courseName>-template
# https://github.com/<organization-shortname>/<courseName>-<studentName>
course_name: my-course

# Absolute path to the course roster (list of students downloaded from
# github classroom)
roster: /Users/me/awesome-course/roster.csv

# Absolute path to the nbgrader directory (formatted as per the
# nbgrader docs).
nbgrader_dir: /Users/me/awesome-course/nbgrader

# Absolute path to directory where you want to clone the student
# repositories. Because we will create many git repos here, this directory
# should not be in a git repo
clone_dir: /Users/me/awesome-course/cloned_dirs

# Absolute path to the assignment template repositories. Again, none of
# the parent directories should be a git repo
template_dir: /Users/me/awesome-course/assignment_repos

# Copy files in the nbgrader 'release' directory to the template repository
# if they match any of these Unix shell-style patterns (NOT regular expressions)
template_patterns:
- *.ipynb

# Additional files that will be created and added to the student repositories
# Each item in the list will be added as a line in the file
extra_files:
  gitignore:
  - .ipynb_checkpoints
  - .DS_Store
  README.md:
  - \nThis repository contains the notebooks for this week's assignment.
  - Complete the homework in each .ipynb notebook, commit your work, and
  - push the changes to github. Your instructor will pull the completed
  - assignment after the deadline.
  - \nOnce grading is complete, your instructor will push the results to a
  - file called `feedback.html`. You will need to pull the changes to your
  - local copy and open this file in a browser to view (github will not
  - render html files).
```

do i have a dated config? can you get in sync with @nkorinek on the contents of that config if i do?